src/NLog/Targets/Target.cs

```diff
@@ -719,6 +721,42 @@ protected string RenderLogEvent(Layout layout, LogEventInfo logEvent)
             }
         }
 
+        /// <summary>
+        /// Renders the event info in layout.
+        /// </summary>
+        /// <param name="layout">The layout.</param>
+        /// <param name="logEvent">The event info.</param>
+        /// <param name="defaultValue">The default value when <paramref name="layout"/> and/or <paramref name="logEvent"/> is <c>null</c></param>
+        /// <returns>Value representing log event.</returns>
+        protected T RenderLogEvent<T>([CanBeNull] Layout<T> layout, [CanBeNull] LogEventInfo logEvent, T defaultValue = default(T))
+        {
+            if (layout == null || logEvent == null)
+                return defaultValue;
+
+            if (OptimizeBufferReuse)
+            {
+                if (layout.IsFixed)
+                {
+                    return layout.RenderableToValue(logEvent, defaultValue);
+                }
+
+                if (TryGetCachedValue(layout, logEvent, out var value))
+                {
+                    return value;
+                }
+
+                return layout.RenderableToValue(logEvent, defaultValue); //todo
+                //using (var localTarget = ReusableLayoutBuilder.Allocate())
```

Comment:
@snakefoot please your advise on this one! Thanks!

Reply:
Well if you can dig out a raw-value then no need to render anything, but else you should provide the StringBuilder to RenderableToValue so it is available if needed.

Reply:
I guess the performance issue is here:

```
 var text = _layout.Render(logEvent);
            if (TryParse(text, out var parsedValue))
            {
                return parsedValue;
            }
```

in TypedLayoutBase. But that code won't get called if the type supported IRawValue

Reply:
Yes instead of calling `_layout.Render` then it should call the version that accepts a pre-allocated StringBuilder.

Reply:
@snakefoot I think I got it. Please check (this PR or just 1714c3de57f4f41bd2b5765ad3a291cd90f9a532). 


I really like to merge this one. 

Thanks!